### DISCUSSION TITLE
Issue Backlog Cleanup

### DISCUSSION THREAD

**Discussion by glanham-jr** [searxng/searxng#3752] [General]
I have some interest in trying to comb through the current suite of issues and try to address some of them. However, there are many which seem to be in an intermediate state, possible duplicates, and some which may not be applicable anymore. Some issues go back 4 yearz

The goal isn't to necessarily just make the number of issues go down, but I was trying to see what would be feasible for me to tackle with my current knowledge of SearxNG. I also don't want to create too much noise reviving dead issues either.

Any thoughts on approaching this as a contributor?

**Comment by return42** [comment]
Wow, thanks a lot for your time and effort :+1: 

Through regular maintenance over the last few years, I have tried to keep the list of issues reasonably up to date.  However, some of them will have to be looked at again today and may have to be discarded or edited.  If we approach the issues from this perspective, then in my opinion there are some groups to consider:

1. issues with [label: bug](https://github.com/searxng/searxng/issues?page=2&q=is%3Aissue+is%3Aopen+label%3Abug) (~50)
2. issues with [label: new feature](https://github.com/searxng/searxng/issues?q=is%3Aissue+is%3Aopen+label%3A%22new+feature%22) (~80)
3. issues with [label: engine request](https://github.com/searxng/searxng/issues?q=is%3Aissue+is%3Aopen+label%3A%22engine+request%22) (~20)
4. issues [without any of the above labels](https://github.com/searxng/searxng/issues?q=is%3Aissue+is%3Aopen+-label%3Abug%2C%22new+feature%22%2C%22engine+request%22+) (~50)

This is the rough structure I usually use to look at the issues.   But I have also tried to summarize some topics and interests , such as:

- https://github.com/searxng/searxng/issues/3176

There are no real tips for further maintenance of the list.

- The bugs are at the top of my list of priorities
- If I had a little more time, I would probably review and cleanup of issues in group 4. There are some topics that may no longer be current, can be closed due to lack of interest or should be labeled.
- https://github.com/searxng/searxng/issues/141
-

**Reply by glanham-jr** [reply]
This generally aligns with how I view it was well. I've been combing through bugs, but some definitely aren't as easy to tackle as others (Bing already me a headache). I'll like my continue focusing on reducing bugs in the meantime, maybe checking for small features.

**Reply by return42** [reply]
> Bing already me a headache

You are not alone .. took me mirades without any progress / M$-bing is just a mess .. started to [tag bing issues](https://github.com/searxng/searxng/labels/bing-engine) a while ago.

> I'll like my continue focusing on reducing bugs in the meantime, maybe checking for small features.

Thats nice .. thanks a lot :+1: .. but keep in mind: Don't waste too much time on things that don't make you happy. This is a free time project and it should be fun to take part in it .. This is also the reason why I personally gave up on bing :-)
